test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/http/HttpResponsePoller.java

```diff
@@ -8,36 +8,63 @@
 
 import uk.gov.justice.services.test.utils.core.rest.RestClient;
 
+import java.util.function.Predicate;
+
 import javax.ws.rs.core.Response;
 
 public class HttpResponsePoller {
 
     private static final int DEFAULT_DELAY_IN_MILLIS = 1000;
     private static final int RETRY_COUNT = 5;
 
-    private final RestClient restClient = new RestClient();
+    private RestClient restClient;
 
-    public String pollUntilNotFound(final String url, final String mediaType) {
+    private HttpResponsePoller(RestClient restClient) {
+        this.restClient = restClient;
+    }
+
+    public static HttpResponsePoller httpResponsePoller() {
+        return new HttpResponsePoller(new RestClient());
```

Comment:
Does this break the backwards compatibility of how everyone else is constructing/using this? - Does it have to?